File: src/main/java/io/github/wntopia/gikipedia/server/domain/auth/dto/internal/DataGsmToken.java
package io.github.wntopia.gikipedia.server.domain.auth.dto.internal;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.io.Serializable;

@JsonIgnoreProperties(ignoreUnknown = true)
public record DataGsmToken(
    @JsonProperty("access_token") String accessToken,
    @JsonProperty("token_type") String tokenType,
    @JsonProperty("expires_in") Long expiresIn,
    @JsonProperty("refresh_token") String refreshToken,
    String scope)
    implements Serializable {}

File: src/main/kotlin/io/github/wntopia/gikipedia/server/domain/auth/dto/internal/DataGsmToken.kt
package io.github.wntopia.gikipedia.server.domain.auth.dto.internal

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import java.io.Serializable

@JsonIgnoreProperties(ignoreUnknown = true)
data class DataGsmToken(
    @JsonProperty("access_token") val accessToken: String,
    @JsonProperty("token_type") val tokenType: String,
    @JsonProperty("expires_in") val expiresIn: Long,
    @JsonProperty("refresh_token") val refreshToken: String,
    val scope: String,
) : Serializable
